lib/Controller/MessagesController.php

```diff
@@ -220,6 +220,55 @@ public function move($accountId, $folderId, $id, $destAccountId, $destFolderId):
 		return new JSONResponse();
 	}
 
+	/**
+	 * @NoAdminRequired
+ 	 * @NoCSRFRequired
+	 * @TrapError
+	 *
+	 * @param int $accountId
+	 * @param string $folderId
+	 * @param int $messageId
+	 *
+	 * @return JSONResponse
+	 * @throws ServiceException
+	 */
+	public function getRawMessage(int $accountId, string $folderId, int $messageId): Response {
+		try {
+			$account = $this->accountService->find($this->currentUserId, $accountId);
+		} catch (DoesNotExistException $e) {
+			return new TemplateResponse(
+				$this->appName,
+				'error',
+				['message' => 'Not allowed'],
+				'none'
+			);
+		}
+
+		$htmlResponse = new HtmlResponse(
+			nl2br(
+				$this->mailManager->getRawMessage(
+					$account,
+					base64_decode($folderId),
+					$messageId
+				)
+			)
+		);
+
+		// Harden the default security policy
+		$policy = new ContentSecurityPolicy();
+		$policy->allowEvalScript(false);
```

Comment:
this is deprecated. we shouldn't use this method. Why do we need a custom CSP here btw?

Reply:
I've just copied what's done in `getHtmlBody()`.

Should I remove this?

Reply:
yes please

Reply:
Ok, done. Should I remove this $policy thing from `getHtmlBody` too?